### ISSUE TITLE
What the meaning of ITER_SIZE and always get stuck during training?

### ISSUE THREAD

**Issue by littlebelly** [kazuto1011/deeplab-pytorch#10]
Hi, When I train voc dataset using your code, it always stuck at some time (when I use 4xM40 to train my model, it always stuck in iter 282/20000), no error no crash, just got stuck, do you have any idea about this error? (miou is about 74 after train about 7 hours, which is reasonable).

and I saw you set the ITER_SIZE in voc12.yaml, what the purpose of this parameter and does it have any  connection with my error?

Thanks in advance!

**Comment by kazuto1011** [comment]
For iter_size, please look at [README](https://github.com/kazuto1011/deeplab-pytorch/blob/master/README.md#default-settings) and [train.py](https://github.com/kazuto1011/deeplab-pytorch/blob/master/train.py#L159-L196). It is equivalent to iter_size in Caffe.
> BVLC/caffe wiki
Accumulate gradients across batches through the iter_size solver field. With this setting batch_size: 16 with iter_size: 1 and batch_size: 4 with iter_size: 4 are equivalent.

The purpose is to save a video memory, although you might be able to set ```BATCH_SIZE: 10, ITER_SIZE: 1``` with your GPUs. I'm not sure about your stucking issue but I have ever seen such reports with the multi-GPU environment which is due to NCCL problem.

**Comment by kazuto1011** [comment]
On my machine, the code has succeeded to train with two Titan X. Just for your information.

**Comment by littlebelly** [comment]
@kazuto1011 
Thanks for your reply! I solved this bug, it caused by opencv in dataloader, you can check the details in this issue: pytorch/pytorch#1355, just add __cv2.setNumThreads(0)__ in  __init__ function of the dataloader.

Thanks for your codes and kind help again! :)
